src/SourceBuild/Arcade/src/Tarball_ReadSourceBuildIntermediateNupkgDependencies.cs

```diff
@@ -137,5 +148,32 @@ XName CreateQualifiedName(string plainName)
 
             return !Log.HasLoggedErrors;
         }
+
+        private string ConvertInternalRepo(string uri)
+        {
+            if (uri.StartsWith("https://dev.azure.com", StringComparison.OrdinalIgnoreCase))
+            {
+                string[] repoParts = uri.Substring(uri.LastIndexOf('/')).Split('-', 2);
+
+                if (repoParts.Length != 2)
+                {
+                    Log.LogError($"Repo '{uri}' does not end with the expected <GH organization>-<GH repo> format");
+                    return null;
+                }
+
+                string org = repoParts[0];
+                string repo = repoParts[1];
+
+                // The internal Nuget.Client repo has suffix which needs to be accounted for.
+                if (uri.EndsWith("-Trusted", StringComparison.OrdinalIgnoreCase))
+                {
+                    repo = repo.Substring(0, repo.Length - 8);
```

Comment:
Personal preference: I like
`repo = repo.Substring(0, repo.Length - "-Trusted".Length);`
since you don't have to worry about counting characters and it shows up if anyone tries to find/replace later.

Reply:
Fixed - I defined a local const for this.